arch/arm/include/cortex_m/stack.h

```diff
@@ -49,6 +49,13 @@ static ALWAYS_INLINE void _InterruptStackSetup(void)
 #endif
 
 	_MspSet(msp);
+#if defined(CONFIG_BUILTIN_STACK_GUARD)
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SPLIM)
+	_MsplimSet((u32_t)_interrupt_stack);
```

Comment:
Any reason not to use __set_MSPLIM from CMSIS?

Reply:
Yes, the problem is that kernel has its own way to define core arm-specific functions, bypassing CMSIS. This is very problematic IMO.
I could fix it for stack.h, and try to include CMSIS. This will require #include <soc.h> because we need soc-speficic defines, e.g. __DSP_PRESENT....

Reply:
@galak See #6568

Reply:
Merged #6568 so you can update this now

Reply:
@galak Rebased :)